src/query/exceptions.hpp

```diff
@@ -211,6 +211,26 @@ class IsolationLevelModificationInMulticommandTxException : public QueryExceptio
       : QueryException("Isolation level cannot be modified in multicommand transactions.") {}
 };
 
+class IsolationLevelModificationInAnalyticsException : public QueryException {
+ public:
+  IsolationLevelModificationInAnalyticsException()
+      : QueryException("Isolation level cannot be modified when storage mode is set to IN_MEMORY_ANALYTICAL.") {}
+};
+
+class StorageModeModificationInMulticommandTxException : public QueryException {
+ public:
+  StorageModeModificationInMulticommandTxException()
+      : QueryException("Storage mode cannot be modified in multicommand transactions.") {}
+};
+
+class StorageModeModificationInMultiTxException : public QueryException {
+ public:
+  StorageModeModificationInMultiTxException()
+      : QueryException(
+            "Storage mode cannot be modified when multiple transactions are running. Check the status of the "
+            "transactions using 'SHOW TRANSACTIONS' query.") {}
```

Comment:
Append: "and ensure no other transactions are active while you attempt to switch the storage mode."